sdk/eventhub/eventhubs-checkpointstore-blob/src/blobCheckpointStore.ts

```diff
@@ -230,25 +238,39 @@ export class BlobCheckpointStore implements CheckpointStore {
     const blobClient = this._containerClient.getBlobClient(blobName);
 
     if (etag) {
-      return await blobClient.setMetadata(metadata as Metadata, {
+      return blobClient.setMetadata(metadata as Metadata, {
         conditions: {
           ifMatch: etag
         }
       });
     } else {
       const blockBlobClient = blobClient.getBlockBlobClient();
 
-      return await blockBlobClient.upload("", 0, {
+      return blockBlobClient.upload("", 0, {
         metadata: metadata as Metadata
       });
     }
   }
 }
 
 type OwnershipMetadata = {
-  [k in "ownerid"]: string;
+  [k in "ownerid"]: string|undefined;
 };
 
 type CheckpointMetadata = {
-  [k in "sequencenumber" | "offset"]: string;
+  [k in "sequencenumber" | "offset"]: string|undefined;
 };
+
+export function parseIntOrThrow(blobName: string, fieldName: string, numStr: string | undefined): number {
+  if (numStr == null) {
+    throw new Error(`Missing metadata property '${fieldName}' on blob '${blobName}'`);
+    }
```

Comment:
Nit: I think you need to run format again :)

Reply:
Done.